### PR TITLE
Remake TriggeringTesla Event

### PR DESCRIPTION
**Pull request by louis1706**
Make what config say

### CODE REVIEW

**Comment by iopietro** [comment]
The name `TriggeringTeslaEventArgs::IsAllowed` is more intuitive and consistent as it is.

**Comment by Misaka-ZeroTwo** [comment]
``IsTriggerable``, ``IsAllowed``, and ``DisableTesla``
Seems to be very confusing

**Comment by louis1706** [comment]
> ``IsTriggerable``, ``IsAllowed``, and ``DisableTesla``
> Seems to be very confusing

What would be better @Misaka-ZeroTwo ?

File: Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs [hunk 1]
```diff
@@ -57,5 +57,15 @@ public TriggeringTeslaEventArgs(Player player, TeslaGate teslaGate)
         ///     Gets the Tesla.
         /// </summary>
         public TeslaGate Tesla { get; }
+
+        /// <summary>
+        ///     Gets or sets a value indicating whether or not the player will be detect by the tesla.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs:62]:
```suggestion
        ///     Gets or sets a value indicating whether or not the player will be detected by the tesla.
```

File: Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs [hunk 2]
```diff
@@ -57,5 +57,15 @@ public TriggeringTeslaEventArgs(Player player, TeslaGate teslaGate)
...
+        /// </summary>
+        public bool IsAllowed { get; set; } = true;
+
+        /// <summary>
+        ///     Gets or sets a value indicating whether or not the tesla is going to be activated.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs:67]:
```suggestion
        ///     Gets or sets a value indicating whether or not the tesla will be deactivated.
```
consistency

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs [hunk 3]
```diff
@@ -39,14 +39,14 @@ public TriggeringTeslaEventArgs(Player player, TeslaGate teslaGate)
...
 
         /// <summary>
-        ///     Gets or sets a value indicating whether or not the is going to be idle.
+        ///     Gets or sets a value indicating whether or not the player will idling the tesla and going to be idle.
         /// </summary>
         public bool IsInIdleRange { get; set; } = true;
 
         /// <summary>
-        ///     Gets or sets a value indicating whether or not the tesla is going to be activated.
+        ///     Gets or sets a value indicating whether or not the player will triggering the tesla and going to be activated.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs:47]:
```suggestion
        ///     Gets or sets a value indicating whether or not the player will cause the tesla going to be activated.
```

File: Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs [hunk 4]
```diff
@@ -39,14 +39,14 @@ public TriggeringTeslaEventArgs(Player player, TeslaGate teslaGate)
         public bool IsInHurtingRange { get; set; }
 
         /// <summary>
-        ///     Gets or sets a value indicating whether or not the is going to be idle.
+        ///     Gets or sets a value indicating whether or not the player will idling the tesla and going to be idle.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Player/TriggeringTeslaEventArgs.cs:42]:
```suggestion
        ///     Gets or sets a value indicating whether or not the player will cause the tesla going to be idle.
```

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Review by Michal78900 — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**